src/components/ReadingProgressPage/ReadingHistory/AddReading/index.tsx

```diff
@@ -0,0 +1,322 @@
+        });
+      })
+      .catch(() => {
+        toast(t('common:error.general'), {
+          status: ToastStatus.Error,
+        });
+      })
+      .finally(() => setIsSubmitting(false));
+  };
+
+  return (
+    <>
+      <Button onClick={onOpenClick} variant={ButtonVariant.Outlined}>
+        {t('manually-add')}
+      </Button>
+      <Modal
+        isOpen={isOpen}
+        onClickOutside={onClose}
+        onEscapeKeyDown={onClose}
+        size={ModalSize.LARGE}
+      >
+        <Modal.Body>
+          <Modal.Header>
+            <Modal.Title>
+              {selectedDate ? (
+                <div className={styles.selectedDateHeaderContainer}>
+                  <Button
+                    size={ButtonSize.Medium}
+                    variant={ButtonVariant.Ghost}
+                    onClick={onGoBackClick}
+                    className={styles.backButton}
+                  >
+                    <ArrowLeft />
+                  </Button>
+                  <p>
+                    {t('history-for', {
+                      date: dateToReadableFormat(selectedDate, lang, {
+                        year: 'numeric',
+                      }),
+                    })}
+                  </p>
+                </div>
+              ) : (
+                <p>{t('manually-add')}</p>
+              )}
+            </Modal.Title>
+          </Modal.Header>
+          {!selectedDate ? (
+            <>
+              <div className={styles.calendarMonthSelector}>
+                <Button
+                  variant={ButtonVariant.Ghost}
+                  shape={ButtonShape.Circle}
+                  onClick={onMonthBackClick}
+                  isDisabled={selectedMonth === 1}
+                >
+                  <ChevronLeft />
+                </Button>
+                <p className={styles.monthName}>{selectedMonthObj.name}</p>
+                <Button
+                  variant={ButtonVariant.Ghost}
+                  shape={ButtonShape.Circle}
+                  onClick={onMonthForwardClick}
+                  isDisabled={selectedMonth === 12}
+                >
+                  <ChevronRight />
+                </Button>
+              </div>
+              <Calendar
+                month={selectedMonthObj?.id as any}
+                year={selectedYear}
+                onDayClick={onDayClick}
+                getIsDayDisabled={getIsDayDisabled}
+              />
+            </>
+          ) : (
+            <>
+              <VerseRangeInput
+                rangeStartVerse={rangeStart}
+                rangeEndVerse={rangeEnd}
+                onRangeChange={onRangeChange}
+              />
+              <Button isDisabled={getIsAddButtonDisabled()} onClick={onAddClick}>
+                {t('add')}
+              </Button>
+
+              <div className={styles.durationInputWrapper}>
+                <DurationInput
+                  totalSeconds={totalSeconds}
+                  onTotalSecondsChange={setTotalSeconds}
```

Comment:
without setting the minimum to 0, a negative number can be submitted
<img width="780" alt="Screen Shot 2023-09-03 at 2 13 00 PM" src="https://github.com/quran/quran.com-frontend-next/assets/15169499/208db418-1949-41e5-9fc2-40ff8796302c">